### DISCUSSION TITLE
Swipe Forward to page, or router.goForward()

### DISCUSSION THREAD

**Discussion by sanctas** [expo/router#678] [Q&A]
So, since this library functions much more like a browser, to me it only makes sense that we should be able to go forward in the navigation like we can go back. 

So, you would be on the “Home” screen, navigate to the “My Profile”  and then (either by swiping from left-to-right or pressing a back button) call router.goBack() and end up back at the “Home” screen. Then you would be able to (by swiping right-to-left) call router.goForward() to go *back* to the “My Profile” screen.

This functionality exists in a lot of native apps I use, like Apollo for Reddit. Here is a video of the behavior:

https://github.com/expo/router/assets/136722448/55e7c961-f003-4eb9-823c-058909d54195

Is such functionality possible, and does anyone have any thoughts about how this might be implemented using the current release? I have poked around and I am stumped.

**Comment by RalfiBahar** [comment]
Any suggestions?
